radixdlt/src/main/java/com/radixdlt/consensus/liveness/RotatingLeaders.java

```diff
@@ -18,26 +18,21 @@
 package com.radixdlt.consensus.liveness;
 
 import com.google.common.collect.ImmutableList;
-import com.radixdlt.common.EUID;
 import com.radixdlt.consensus.View;
+import com.radixdlt.crypto.ECPublicKey;
 import java.util.Objects;
 
 /**
  * Round Robin leaders
  */
 public final class RotatingLeaders implements ProposerElection {
-	private final ImmutableList<EUID> leaders;
-	public RotatingLeaders(ImmutableList<EUID> leaders) {
+	private final ImmutableList<ECPublicKey> leaders;
+	public RotatingLeaders(ImmutableList<ECPublicKey> leaders) {
 		this.leaders = Objects.requireNonNull(leaders);
 	}
 
 	@Override
-	public boolean isValidProposer(EUID nid, View view) {
-		return nid.equals(getProposer(view));
-	}
-
-	@Override
-	public EUID getProposer(View view) {
+	public ECPublicKey getProposer(View view) {
 		int index = (int) (view.number() % leaders.size());
```

Comment:
Appreciate that this isn't part of this PR, but I just noticed that it would be good to note that `view.number()` is guaranteed to be positive here (otherwise we might run into issues using it as an index since Java uses remainder for modulus on negative numbers).